docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+* So, if an app has an ad-hoc design of how to disseminate transactions,
+  there is no point in having transactions in CometBFT's mempool.
+
+This proposal targets this kind of applications:
+those that have an ad-hoc (and likely more efficient) mechanism for transaction dissemination.
+We propose to introduce a `nil` mempool which will effectively act as a stubbed object
+internally:
+
+* it will reject any transaction being locally submitted or gossipped by a peer
+* when it is time to _reap_ (as it is currently called) the mempool the answer will always be empty
+* the application running on the proposer validator will add transactions it received
+  using the appchains's own mechanism via `PrepareProposal`.
+
+## Alternative Approaches
+
+These are the alternatives known to date:
+
+1. Keep the current model. Useful for basic apps, but clearly suboptimal for applications
+   with their own mechanism to disseminate transactions and particular performance requirements.
+2. Provide more efficient general-purpose mempool implementations.
+   This is an ongoing effort, but will take some time (and R&D effort) to come up with
+   advanced mechanisms -- likely highly configurable -- which then will have to be thoroughly tested.
+3. A similar approach to this one ([ADR110][adr-110]) whereby the application-specific
+   mechanism directly interacts with CometBFT via a newly defined gRPC interface.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+What this ADR proposes can already be achieved with an unmodified CometBFT since `v0.37.x`,
+albeit with a complex, poor UX.
+This core of this proposal is to make some internal changes so it is clear an simple for app developers,
+thus improving the UX.
+
+#### `nil` Mempool
+
+We propose a new mempool implementation, called `nil` Mempool, that effectively disables all mempool functionality
+within CometBFT.
+The `nil` Mempool implements the `Mempool` interface in a very simple manner:
+
+*	`CheckTx(tx types.Tx) (*abcicli.ReqRes, error)`: returns `nil, ErrNotAllowed`
+*	`RemoveTxByKey(txKey types.TxKey) error`: returns `ErrNotFound`
+* `ReapMaxBytesMaxGas(maxBytes, maxGas int64) types.Txs`: returns `nil`
+* `ReapMaxTxs(max int) types.Txs`: returns `nil`
+*	`Lock()`: does nothing
+* `Unlock()`: does nothing
+*	`Update(...) error`: returns `nil`
+* `FlushAppConn() error`: returns `nil`
+*	`Flush()`: does nothing
+* `TxsAvailable() <-chan struct{}`: returns a closed channel
+* `EnableTxsAvailable()`: does nothing
+* `SetTxRemovedCallback(cb func(types.TxKey))`: does nothing
+* `Size() int` returns 0
+* `SizeBytes() int64` returns 0
+
+Upon startup, the `nil` mempool reactor will advertise no channels to the peer to peer layer.
+
+### Configuration
+
+We propose the following changes to the `config.toml` file:
+
+```toml
+[mempool]
+# The type of mempool for this CometBFT node to use.
+#
+# Valid types of mempools supported by CometBFT:
+# - "flood" : clist mempool with flooding gossip protocol (default)
```

Comment:
local (or a similar word) would be my preference because

a) I doubt that CometBFT will have more than one mempool implementation in the future,
b) devs/operators usually don't know/care about the concrete mechanics (either it's point-to-point, gossip, or something else). I.e., I see some operators confused about flood option if they didn't read the docs

Reply:
Not fully convinced (we might at CAT mempool land in the next weeks/months) but will change it to "local" as you folks seem to have a strong opinion on this one.

Reply:
> The type of mempool is not the same as the gossip protocol used by the mempool (if any)

💯 

> Not fully convinced (we might have CAT mempool landing on `main` in the next weeks/months) but will change it to "local" as you folks seem to have a strong opinion on this one.

Hmm, in that case, let's stick with `flood` or `clist` (`type CListMempool struct`) because of ^